environs/interface.go

```diff
@@ -292,6 +298,7 @@ type Bootstraper interface {
 type Configer interface {
 	ConfigGetter
 	ConfigSetter
+	CloudSpecSetter
```

Comment:
Does this need to be added here? Could you not check if the environ implements the `CloudSpecSetter` and then call the method. That way we don't need the empty `SetCloudSpec` in every provider?